_posts/2019-06-25-dynamic_linear_models_tfprobability/dynamic_linear_models_tfprobability.Rmd

```diff
@@ -0,0 +1,805 @@
+
+```{r}
+ggplot(forecast_df, aes(x = month, y = smoothed_means_intercept)) +
+  geom_line(color = "orange") +
+  geom_line(aes(y = smoothed_means_slope),
+            color = "green") +
+  geom_ribbon(
+    aes(
+      ymin = smoothed_means_intercept - 2 * smoothed_sds_intercept,
+      ymax = smoothed_means_intercept + 2 * smoothed_sds_intercept
+    ),
+    alpha = 0.3,
+    fill = "orange"
+  ) +
+  geom_ribbon(
+    aes(
+      ymin = smoothed_means_slope - 2 * smoothed_sds_slope,
+      ymax = smoothed_means_slope + 2 * smoothed_sds_slope
+    ),
+    alpha = 0.1,
+    fill = "green"
+  ) +
+  coord_cartesian(xlim = c(forecast_df$month[1], forecast_df$month[length(ts) - n_forecast_steps]))  +
+  theme(axis.title = element_blank())
+
+```
+
+
+```{r, eval=TRUE, echo=FALSE, layout="l-body-outset", fig.cap = "Smoothing estimates from the Kálmán filter. Green: coefficient for dependence on excess market returns (slope), orange: vector of ones (intercept)."}
+knitr::include_graphics("images/capm_smoothed.png")
+```
+
+
+
+#### Filtering estimates
+
+For comparison, here are the filtering estimates. Note that here, the y-axis extends further up and down so we can capture uncertainty better: ^[It is still clipped up and below though. We've been trying to find a compromise between local visibility of the filtered means on the one hand, and global visibility of uncertainty in the estimates on the other.]
+
+```{r}
+ggplot(forecast_df, aes(x = month, y = filtered_means_intercept)) +
+  geom_line(color = "orange") +
+  geom_line(aes(y = filtered_means_slope),
+            color = "green") +
+  geom_ribbon(
+    aes(
+      ymin = filtered_means_intercept - 2 * filtered_sds_intercept,
+      ymax = filtered_means_intercept + 2 * filtered_sds_intercept
+    ),
+    alpha = 0.3,
+    fill = "orange"
+  ) +
+  geom_ribbon(
+    aes(
+      ymin = filtered_means_slope - 2 * filtered_sds_slope,
+      ymax = filtered_means_slope + 2 * filtered_sds_slope
+    ),
+    alpha = 0.1,
+    fill = "green"
+  ) +
+  coord_cartesian(ylim = c(-2, 2),
+                  xlim = c(forecast_df$month[1], forecast_df$month[length(ts) - n_forecast_steps])) +
+  theme(axis.title = element_blank())
+```
+
+```{r, eval=TRUE, echo=FALSE, layout="l-body-outset", fig.cap = "Filtering estimates from the Kálmán filter. Green: coefficient for dependence on excess market returns (slope), orange: vector of ones (intercept)."}
+knitr::include_graphics("images/capm_filtered.png")
+```
+
+
+Summing up so far, by now we've seen a full example of time series fitting, forecasting, and smoothing/filtering, in an exciting setting one doesn't encounter too often: dynamic linear regression. What we haven't seen as yet is the additivity feature of DLMs, and how it allows us to _decompose_ a time series into its (theorized) constituents.
```

Comment:
```suggestion
So far, we've seen a full example of time-series fitting, forecasting, and smoothing/filtering, in an exciting setting one doesn't encounter too often: dynamic linear regression. What we haven't seen as yet is the additivity feature of DLMs, and how it allows us to _decompose_ a time series into its (theorized) constituents.
```

Reply:
I'm trying to take out some of the opening qualifiers, when you have more than one.